File: SYP.Web/Modules/ServerTypes/Email/SmtpSettingsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SmtpSettingsRow } from "./SmtpSettingsRow";

export interface SmtpSettingsColumns {
    Id: Column<SmtpSettingsRow>;
    Name: Column<SmtpSettingsRow>;
    Host: Column<SmtpSettingsRow>;
    Port: Column<SmtpSettingsRow>;
    UseSsl: Column<SmtpSettingsRow>;
    FromAddress: Column<SmtpSettingsRow>;
    IsDefault: Column<SmtpSettingsRow>;
    IsActive: Column<SmtpSettingsRow>;
}

export class SmtpSettingsColumns extends ColumnsBase<SmtpSettingsRow> {
    static readonly columnsKey = 'Email.SmtpSettings';
    static readonly Fields = fieldsProxy<SmtpSettingsColumns>();
}

File: SYP.Web/Modules/Email/SmtpSettings/SmtpSettingsColumns.cs
namespace SYP.Email.Columns;

[ColumnsScript("Email.SmtpSettings")]
[BasedOnRow(typeof(SmtpSettingsRow), CheckNames = true)]
public class SmtpSettingsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight, Width(80)]
    public int Id { get; set; }

    [EditLink, Width(200)]
    public string Name { get; set; }

    [Width(200)]
    public string Host { get; set; }

    [Width(80)]
    public int Port { get; set; }

    [Width(80)]
    public bool UseSsl { get; set; }

    [Width(200)]
    public string FromAddress { get; set; }

    [Width(80)]
    public bool IsDefault { get; set; }

    [Width(80)]
    public bool IsActive { get; set; }
}
